track_info.c

```diff
@@ -46,8 +46,12 @@ static void track_info_free(struct track_info *ti)
 
 struct track_info *track_info_new(const char *filename)
 {
+	static uint64_t cur_uid = 0;
+	cur_uid++;
```

Comment:
If you remove a track, free the track_info, add another track, allocate a new track_info, then that track_info can have the same address as the old one. A 64 bit integer cannot realistically overflow.